cmd/livepeer_bench/livepeer_bench.go

```diff
@@ -0,0 +1,228 @@
+
+	flag.Parse()
+
+	if *in == "" {
+		glog.Errorf("Please provide the input manifest as `%s -in <input.m3u8>`", os.Args[0])
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	profiles := parseVideoProfiles(*transcodingOptions)
+
+	f, err := os.Open(*in)
+	if err != nil {
+		glog.Fatal("Couldn't open input manifest: ", err)
+	}
+	p, _, err := m3u8.DecodeFrom(bufio.NewReader(f), true)
+	if err != nil {
+		glog.Fatal("Couldn't decode input manifest: ", err)
+	}
+	pl, ok := p.(*m3u8.MediaPlaylist)
+	if !ok {
+		glog.Fatalf("Expecting media playlist in the input %s", *in)
+	}
+
+	accel := ffmpeg.Software
+	devices := []string{}
+	if *nvidia != "" {
+		accel = ffmpeg.Nvidia
+		devices = strings.Split(*nvidia, ",")
+	}
+
+	ffmpeg.InitFFmpeg()
+	segCount := 0
+	var wg sync.WaitGroup
+	dir := path.Dir(*in)
+	start := time.Now()
+
+	table := tablewriter.NewWriter(os.Stderr)
+	data := [][]string{
+		{"Source File", *in},
+		{"Transcoding Options", *transcodingOptions},
+		{"Concurrent Sessions", fmt.Sprintf("%v", *concurrentSessions)},
+	}
+
+	for _, v := range data {
+		table.Append(v)
+	}
+
+	table.SetAlignment(tablewriter.ALIGN_LEFT)
+	table.SetCenterSeparator("*")
+	table.SetColumnSeparator("|")
+	table.Render()
+
+	fmt.Println("timestamp,session,segment,transcode_time")
+	for i := 0; i < *concurrentSessions; i++ {
+		wg.Add(1)
+		go func(k int, wg *sync.WaitGroup) {
+			tc := ffmpeg.NewTranscoder()
+			for j, v := range pl.Segments {
+				if *segs > 0 && j >= *segs {
+					break
+				}
+				if v == nil {
+					continue
+				}
+				u := path.Join(dir, v.URI)
+				in := &ffmpeg.TranscodeOptionsIn{
+					Fname: u,
+					Accel: accel,
+				}
+				if ffmpeg.Software != accel {
+					in.Device = devices[k%len(devices)]
+				}
+				profs2opts := func(profs []ffmpeg.VideoProfile) []ffmpeg.TranscodeOptions {
+					opts := []ffmpeg.TranscodeOptions{}
+					for n, p := range profs {
+						oname := ""
+						if *outPrefix != "" {
+							oname = fmt.Sprintf("%s_%s_%d_%d_%d.ts", *outPrefix, p.Name, n, k, j)
+						} else {
+							oname = "-"
+						}
+						o := ffmpeg.TranscodeOptions{
+							Oname:        oname,
+							Profile:      p,
+							Accel:        accel,
+							AudioEncoder: ffmpeg.ComponentOptions{Name: "copy"},
+							Muxer:        ffmpeg.ComponentOptions{Name: "mpegts"},
+						}
+						opts = append(opts, o)
+					}
+					return opts
+				}
+				out := profs2opts(profiles)
+				t := time.Now()
+				_, err := tc.Transcode(in, out)
+				end := time.Now()
+				fmt.Printf("%s,%d,%d,%0.4v\n", end.Format("2006-01-02 15:04:05.9999"), k, j, end.Sub(t).Seconds())
+				if err != nil {
+					glog.Fatalf("Transcoding failed for session %d segment %d: %v", k, j, err)
+				}
+				segCount++
+			}
+			tc.StopTranscoder()
+			wg.Done()
+		}(i, &wg)
+		time.Sleep(300 * time.Millisecond)
+	}
+	wg.Wait()
+	fmt.Fprintf(os.Stderr, "Took %0.4v seconds to transcode total %v segments (%v concurrent sessions)\n",
```

Comment:
Think the total duration of the segments is useful information since this line would then tell a user that X seconds of video was transcoded in Y seconds which is helpful context since the transcoding speed will depend on the duration of the inputs.

Reply:
Done, also now printing the segment count in the stream, rather than sum of all sessions.